fetchy/extractors/vaccinationDataPerDhbExtrator.ts

```diff
@@ -1,19 +1,65 @@
+import { DateTime } from "luxon";
 import {
   DhbName,
   DhbVaccineDoseData,
 } from "../../shared/types/VaccineDataTypes";
 import { convertToNumber } from "../utilities";
 
 export const getVaccinationDataPerDhb = (
+  dailyUpdatedNzTimeFromHtml: DateTime,
   rawHtml: string
 ): DhbVaccineDoseData[] => {
-  const vaccinationTo90PerDhbTableHtml =
-    /\<a id=\"dhbtables all ethnicities" name="dhbtables all ethnicities\"\>\<\/a\>.*?\<\/h4\>.*?\<tbody\>(.*?)\<\/tbody\>/gs.exec(
-      rawHtml
-    )![1];
+  const rawHtmlDate = new Date(
+    dailyUpdatedNzTimeFromHtml.year,
+    dailyUpdatedNzTimeFromHtml.month,
+    dailyUpdatedNzTimeFromHtml.day
+  );
+
+  if (rawHtmlDate <= new Date(2022, 2, 15)) {
+    return extractDhbData_Including_and_Before_15_02_22(rawHtml);
+  }
+
+  if (rawHtmlDate <= new Date(2022, 2, 16)) {
```

Comment:
Broadly, I like this approach - I think though that we should copy _the whole file_ - RN we know that the regex and the shape of the table have changed. In the future it could be more, and we'll have to come through and refactor this file again. Conversely, if we copy the whole file per date range, we can 1. avoid the `null` checks 2. tie the regex to the extractor. HMU if you want

Reply:
hahahaha I actually did it this way first. You mean like a strategy pattern right?

I assumed people wouldn't want duplication so I put it all in one place even though I prefer strategy pattern too haha. 

Will do that :)

Reply:
If that's what it's called ;) haha yeh I think the duplication is fine :D